### PR TITLE
feat: adds the ability to test against a specific commit-sha

### PR DESCRIPTION
**Pull request by bcoe**
while trying to recreate the issues discussed in: https://github.com/nodejs/node/issues/6456#issuecomment-219255748

I found it useful to smoke test against specific commit shas.

### CODE REVIEW

**Comment by TheAlphaNerd** [comment]
The current approach only support including the sha in the lookup table. It would be nice if this option could be exported as a cli flag as well

**Comment by bcoe** [comment]
@TheAlphaNerd 👍 I will make an effort to update this tonight.

**Comment by TheAlphaNerd** [comment]
awesome thanks! I'll work on getting it landed tomorrow

File: lib/lookup.js [hunk 1]
```diff
@@ -77,7 +83,8 @@ function resolve(context, next) {
           getRepo(rep.repo, meta),
           rep.master ? null : detail.spec,
           meta['dist-tags'],
-          rep.master ? null : rep.prefix);
+          rep.master ? null : rep.prefix,
+          context.options.sha || rep.sha);
```
Comment (TheAlphaNerd) [inline lib/lookup.js]: would you be able to include a test that has coverage of the `context.options.sha` code path

**Comment by bcoe** [comment]
@TheAlphaNerd done 👍

**Comment by TheAlphaNerd** [comment]
LGTM

**Comment by TheAlphaNerd** [comment]
landed in a69427c...5d049e1

**Pull request closed without merging**